### ISSUE TITLE
[Enhancement]: Support PostgreSQL 15

### ISSUE THREAD

**Issue by mkindahl** [timescale/timescaledb#3985]
### What type of enhancement is this?

Other

### What subsystems and features will be improved?

Platform/OS

### What does the enhancement do?

Add support for PostgreSQL 15.

- [ ] Make TimescaleDB build with PostgreSQL 15 source
- [ ] Handle ACL when bootstrapping data node

### Implementation challenges

At the time of filing this issue, PostgreSQL 15 is still in progress so we should be careful about in what order we handle compatibilities. We should first focus on areas that are "stable" (in the sense that we do not expect changes in the code before the release), but wait with "unstable" parts of the code (where we suspect that there can be more changes).

**Comment by fingoldo** [comment]
Hi, PostgreSQL 15 is in Beta 3 already, any news on when Timescale will be compatible?

**Comment by fabriziomello** [comment]
> Hi, PostgreSQL 15 is in Beta 3 already, any news on when TimescaleDB will be compatible?

We're [working on this already](https://github.com/timescale/timescaledb/pulls?q=is%3Apr+label%3Apg15+is%3Aclosed) and trying our best to make it available on the next release.

**Comment by ivanfeanor** [comment]
Hi. Do you have any estimations on the next release?

**Comment by jirutka** [comment]
Are you aware of the problem this poses for Linux distributions? Alpine Linux v3.17 is gonna be released tomorrow, the default PostgreSQL version is 15 and now I have to either remove TimescaleDB from the v3.17, or apply unreleased patches to add support for PostgreSQL 15.

**Comment by jeffreymonroe** [comment]
Any update on this? Perhaps a speculation on when the support for PG15 will be available?

**Comment by fboerman** [comment]
I installed postgresql15 from arch linux AUR and then compiled timescaledb 2.9.1 myself and did a couple of quick tests. so far it works flawlessly! 

All open stuff from above link for psql15 are already merged so give it a go! No need to wait for the official release for it. 

@jeffreymonroe @ivanfeanor @jirutka

**Comment by cbandy** [comment]
@fboerman I think maybe you were looking at the following, which shows only _closed pull requests_:

> We're [working on this already](https://github.com/timescale/timescaledb/pulls?q=is%3Apr+label%3Apg15+is%3Aclosed) and trying our best to make it available on the next release.

This link might be what you're looking for: https://github.com/timescale/timescaledb/labels/pg15

**Comment by fboerman** [comment]
Hi Chris! You are right I think haha. However I have run it for some time now and inserting 500k rows every day in production and have encountered no issues so far.

-------- Original Message --------

**Comment by IsaackRasmussen** [comment]
Documentation appears to be partially updated now for Postgresql-15 support.

https://docs.timescale.com/install/latest/self-hosted/installation-linux/

notable this line: `apt install timescaledb-2-postgresql-15`

**Comment by jirutka** [comment]
You’re right, TimescaleDB 2.9.2 indeed supports PostgreSQL 15, they just didn’t communicate it clearly. I’m at a conference with a few guys from Timescale right now, so I’m gonna talk to them about it.

**Comment by ivanfeanor** [comment]
and there is an official timescaledb-ha image with postgis and pg 15, was waiting for that

**Comment by jirutka** [comment]
I've updated postgresql-timescaledb package in Alpine Linux v3.17 (latest-stable) to the latest version and rebuilt against postgresql15.

**Comment by k7daniel** [comment]
> 

This currently still listed as `apt install timescaledb-2-postgresql-14` still as of posting.

**Comment by hnakamur** [comment]
Release note for [2.10.0](https://github.com/timescale/timescaledb/releases/tag/2.10.0) says "Full PostgreSQL 15 support for all existing features".
However at Step 6 Install TimescaleDB: on https://docs.timescale.com/self-hosted/latest/install/installation-linux/ it says `apt install timescaledb-2-postgresql-14`.

I found `timescaledb-2-postgresql-15` package is in the repository.
```
$ apt show timescaledb-2-postgresql-15 2>/dev/null | head -2
Package: timescaledb-2-postgresql-15
Version: 2.11.0~ubuntu22.04
```

Is the Step 6 on https://docs.timescale.com/self-hosted/latest/install/installation-linux/ needed to be updated?
Or is there something that timescaledb-2-postgresql-14 is better than timescaledb-2-postgresql-15 as of now?

**Comment by pgloader** [comment]
> How to upgrade TimescaleDB 2.5.1 with PostgreSQL 14 to PostgreSQL 15?

It seems to have to have the TimescaleDB upgraded before PostgresQL upgrade

**Comment by svenklemm** [comment]
PostgreSQL 15 support is available for quite a while.
